draft-vvv-webtransport-quic.md

```diff
@@ -129,23 +131,70 @@ the client receives a TLS Finished message from the server.
 
 In order to identify itself as a WebTransport application, QuicTransport relies
 on TLS Application-Layer Protocol Negotiation {{!RFC7301}}.  The user agent MUST
-request the ALPN value of "wq" and it MUST NOT establish the session unless that
-value is accepted.
+request the ALPN value of "wq-draft01" and it MUST NOT establish the session
+unless that value is accepted.
+
+## Client Indication
+
+In order to verify that the client's origin is allowed to connect to the server
+in question, the user agent has to communicate the origin to the server.  This
+is accomplished by sending a special handshake message, called *client
+indication*, on stream 2, which is the first client-initiated unidirectional
+stream.
+
+The client indication is a sequence of key-value pairs that are formatted in the
+following way:
+
+~~~~~~~~~~ drawing
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                           Key (2)                           ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                          Length (2)                         ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                           Value ( )                         ...
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+~~~~~~~~~~
+{: #fig-frame title="Client indication format" :}
+
+The pair includes the following fields:
+
+  Key:
+  : Indicates the field that is being expressed.
+
+  Length:
+  : Indicates the length of the value (the length of the key and the length
+    itself are not included).
 
-## Verifying the Origin
+  Value:
+  : The value of the field, the semantics of which are determined by the key.
+
+The total length of the client indication MUST NOT exceed 65,535 bytes.  The
+keys MAY be duplicate, depending on the definition of the field.  The server
```

Comment:
It's probably simplest to ban duplicate keys like QUIC transport parameters, as this flexibility adds risk

Reply:
Done.